### PR TITLE
feat: add subdomain redirection

### PR DESCRIPTION
**Pull request by tomicvladan**
Fixes the google redirection path issue and supports subdomains.

### CODE REVIEW

File: src/background/listener/bee-api.listener.ts [hunk 1]
```diff
@@ -130,6 +134,23 @@ export class BeeApiListener {
...
+    // NOTE: works only if google search is set as default search engine
+    chrome.webRequest.onBeforeRequest.addListener(
+      (details: chrome.webRequest.WebRequestBodyDetails) => {
+        console.log('Original BZZ Url', details.url)
+        const urlParams = new URLSearchParams(new URL(details.url).search)
+        const query = decodeURI(urlParams.get('oq') || urlParams.get('q') || '')
+
+        if (!query || !query.startsWith('bzz://')) return
+
+        this.redirectToBzzReference(query.substr(6), details.tabId)
```
Comment (nugaon) [inline src/background/listener/bee-api.listener.ts:147]: `string.substr` is a deprecated function

File: src/background/listener/bee-api.listener.ts [hunk 2]
```diff
@@ -157,122 +178,162 @@ export class BeeApiListener {
...
+        id: BeeApiListener.BZZ_GOOGLE_REDIRECT_ID,
+        priority: 2,
+        condition: {
+          regexFilter: isHostLocalhost ? bzzSubdomainGoogleRedirectRegex : bzzGoogleRedirectRegex,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.REDIRECT,
+          redirect: {
+            regexSubstitution: isHostLocalhost ? `${protocol}://\\1.swarm.${host}\\2` : `${this._beeApiUrl}/bzz/\\1`,
```
Comment (nugaon) [inline src/background/listener/bee-api.listener.ts:284]: the subdmain should be in CID form and the bzz path call does not append the original url's path.

**Review by nugaon — CHANGES_REQUESTED** [review]
If I'm adding params to the url then the subdomain redirection does not work:
entering 
https://bah5acgzag4smsbhck2y6sstekmzo43z7buwfc5l7damw2cxjn4yaxr53ok3a.bzz.link/#920a53d48b97006452afbe55dfb293b4458ea2410770b30589cce7355b604309
OR
bzz://3724c904e256b1e94a645332ee6f3f0d2c51757f18196d0ae96f300bc7bb72b6#920a53d48b97006452afbe55dfb293b4458ea2410770b30589cce7355b604309
to the address bar resulting in
http://127.0.0.1:1633/bzz/3724c904e256b1e94a645332ee6f3f0d2c51757f18196d0ae96f300bc7bb72b6/#920a53d48b97006452afbe55dfb293b4458ea2410770b30589cce7355b604309

File: src/background/listener/bee-api.listener.ts (hunk 1 above)
Comment (tomicvladan) [reply src/background/listener/bee-api.listener.ts:147]: This is old code, copied.

File: src/background/listener/bee-api.listener.ts (hunk 2 above)
Comment (tomicvladan) [reply src/background/listener/bee-api.listener.ts:284]: In these declarative web requests, there is no way for any kind of conversion. How the redirection URL should look like?

**Review by nugaon — APPROVED** [review]

**Pull request merged**